pkg/controller/statefulset/stateful_set_utils.go

```diff
@@ -506,3 +506,12 @@ func filterOutCondition(conditions []apps.StatefulSetCondition, condType apps.St
 func getStatefulSetKey(o metav1.Object) string {
 	return o.GetNamespace() + "/" + o.GetName()
 }
+
+func decreaseAndThrottleForMaxUnavailable(maxUnavailable *int) bool {
```

Comment:
i cannot understand what the function returns through the function interface. Consider renaming to decreaseAndCheckMaxUnavailable